### PR TITLE
Index page listing for UserTags per issue #2741.

### LINKED ISSUE #2741: index page or listing of different UserTags
We have a great new UserTag system -- 

https://github.com/publiclab/plots2/blob/master/app/controllers/user_tags_controller.rb

where people can put tags on their profiles. we also have a way to see people by their user tags: 

https://publiclab.org/people/organizer shows everyone with the "organizer" tag.

But we don't have a way to see highly-used user tags, the way we do for regular node tags on this page: 

https://publiclab.org/tags

We should make an `index` method on the `user_tags_controller.rb` and point to it with:

https://publiclab.org/groups (at least for now) in `config/routes.rb`

It can have almost the same styling as the existing /tags page. 

This should be pretty easy! Any help appreciated!

### PR DESCRIPTION
**Pull request by Deep-Red**
Fixes #2741

### CODE REVIEW

**Comment by Deep-Red** [comment]
I'm not certain, but I think the continuous-integration test is designed to fail on this pull request. If I'm wrong and there is an issue I need to address, please let me know. Any additional feedback from @publiclab/reviewers would be greatly appreciated.

**Comment by jywarren** [comment]
Hi! I retriggered it since we had a CI bug just resolved. Thanks a lot for your contribution!!! Let's see... Not sure but I think it should pass. If you'd like to add a functional test for this page, if be happy to help! Take a look in test/functional/user_tags_test.rb for examples!

**Comment by jywarren** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">@Deep-Red Thank you for your pull request! I’m here to help with some tips and recommendations. Please take a look at the list provided and help us review and accept your contribution! And <strong>don’t be discouraged if you see errors</strong> – we’re here to help.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by jywarren** [comment]
These look great! Ping me if you need any help!

**Comment by Deep-Red** [comment]
I'm planning to finish this up tomorrow, I'll probably have a couple of questions for you then @jywarren. Thanks for the support.

**Comment by jywarren** [comment]
awesome! I'll be out away from a computer a lot of tomorrow but will check
in after -- in the meantime, can you request a review from
@publiclab/reviewers ? I'm not 100% sure GitHub lets you request one, but
it should be in the upper right.

**Comment by Deep-Red** [comment]
@jywarren  
I don't think I can request a review from @publiclab/reviewers since I am not listed as a 'contributor' for this repository.  
I still have a lot to learn about writing tests, so I would appreciate a little guidance on the following (in user_tags_controller_test.rb):  

Line 114: This should be testing that the .group(:value) call (user_tags_controller.rb) is working, correct?  
Line 115: This is testing the function of .select('value'), correct?  
Line 116: This test is designed to ensure that only valid data will pass, by checking that a value for which a user_tag does not exist is not present in the collection, correct?  
Line 121: So this should be searching for a valid user_tag. I used one of the user_tags from test/fixtures/user_tags.yml, but it does not seem to be passing the build checks. Why is that?

File: test/functional/user_tags_controller_test.rb [hunk 1]
```diff
@@ -106,4 +106,23 @@ def setup
...
+
+    assert:success
+    assert_equal assigns['user_tags'].sort_by(&:value).sort_by(&:id), assigns['user_tags'].sort_by(&:id)
+    assert_equal assigns['user_tags'].collect(&:value), assigns['user_tags'].collect(&:value).uniq
+    assert_not assigns['user_tags'].collect(&:value).include?(0)
+    assert_not_nil :user_tags
+  end
+
+  test 'user tags search' do
+    get :index, :search => "skill:rails"
```
Comment (jywarren) [inline test/functional/user_tags_controller_test.rb]: Here, it'd be great to use the more modern `, search: "skill:rails"` format.

File: test/functional/user_tags_controller_test.rb [hunk 2]
```diff
@@ -106,4 +106,23 @@ def setup
       assert_equal I18n.t('user_tags_controller.tag_doesnt_exist'), flash[:error]
     end
   end
+
+  test 'user tags index' do
+    get :index
+
+    assert:success
+    assert_equal assigns['user_tags'].sort_by(&:value).sort_by(&:id), assigns['user_tags'].sort_by(&:id)
```
Comment (jywarren) [inline test/functional/user_tags_controller_test.rb]: Here, I think you want `.collect(&:value)` instead of `.sort_by(&:value)`

**Comment by jywarren** [comment]
OK -- apologies, it looks like the new Rails 5 code has caused some merge conflicts. That's one of the largest PRs we've ever merged, so I must apologize for it interrupting your work. It is very unlikely to happen again!

It looks like you're working through them, though, so please know that I can help out if you get stuck!

**Comment by jywarren** [comment]
This looks fantastic! Great work!!!

**Comment by jywarren** [comment]
If you're looking for a new challenge, please take a look at our `help-wanted` list: https://github.com/publiclab/plots2/labels/help-wanted

We'd love your help!! Thanks!

**Pull request merged**